awsshell/wizard.py

```diff
@@ -0,0 +1,141 @@
+import json
+import jmespath
+import botocore.session
+from botocore import xform_name
+
+
+class WizardException(Exception):
+    pass
+
+
+class Wizard(object):
+
+    def __init__(self, filename=None):
+        self._env = Environment()
+        if filename:
+            self.load_from_json(filename)
+
+    # Loads the wizards from the spec in dict form
+    def load_from_dict(self, spec):
+        self.start_stage = spec.get('StartStage', None)
+        if not self.start_stage:
+            raise WizardException("Start stage not specified")
+        self.stages = {}
+        for s in spec['Stages']:
+            stage = Stage(s, self._env)
+            self.stages[stage.name] = stage
+
+    # TODO temporary loader from file, will likely deprecate this when
+    # there's a systemic way of loading all wizards in specified locations
+    def load_from_json(self, filename):
+        with open(filename, 'r') as f:
+            spec = json.load(f)
+            self.load_from_dict(spec)
+
+    # Performs the basic loop for progressing stages
+    def execute(self):
+        current_stage = self.start_stage
+        while current_stage:
+            stage = self.stages.get(current_stage, None)
+            if not stage:
+                raise WizardException("Stage not found: %s" % current_stage)
+            stage.execute()
+            current_stage = stage.get_next_stage()
+        print(self._env)
+
+
+class Stage(object):
+
+    def __init__(self, spec, env):
+        self._wizard_env = env
+        self.name = spec.get('Name', None)
+        self.prompt = spec.get('Prompt', None)
+        self.retrieval = spec.get('Retrieval', None)
+        self.next_stage = spec.get('NextStage', None)
+        self.resolution = spec.get('Resolution', None)
+        self.interaction = spec.get('Interaction', None)
+
+    def __handle_static_retrieval(self):
+        return self.retrieval.get('Resource')
+
+    def __handle_request_retrieval(self):
+        # TODO very basic requests... refactor needed
+        req = self.retrieval['Resource']
+        # initialize botocore session and client for service in the request
+        session = botocore.session.get_session()
```

Comment:
The session should probably live in the Wizard. Clients should probably get a [`CachedClientCreator`](https://github.com/awslabs/aws-shell/blob/master/awsshell/resource/index.py#L140), which also lives in the Wizard.